rse-project.Rmd

```diff
@@ -21,7 +21,214 @@ small pieces in predictable places with readable names are easier to find and us
+And if the list of people changes slowly over time but the meetings continue,
+that's an even stronger sign.
+
+## What files should every project contain? {#project-boilerplate}
+
+TODO: What are the bare minimums here?  I'd say must have `README`, even for your own projects that only live locally.  As soon as something is on GitHub must have `LICENSE` and `CONDUCT`?
+
+Most projects' repositories contain a few files that weren't mentioned in Noble's paper,
+but which have become conventional in open source projects.
+All of these files may be plain text or Markdown,
+and may have a `.txt` or `.md` suffix (or no suffix at all),
+but please use the principal names given,
+in all caps,
+since a growing number of tools expect them.
+
+-   `README`:
+    the project's title and a one-paragraph description of its purpose or content.
+    This file is displayed by GitHub Pages as the project's home page.
+
+-   `LICENSE`:
+    the project's license (discussed in Chapter \@ref(inclusive)).
+
+-   `CONDUCT`:
+    its code of conduct
+    (also discussed in Chapter \@ref(inclusive)).
+
+-   `CITATION`:
+    how the work should be cited.
+    This file usually contains a plain text citation,
+    and may also include entries formatted for various bibliographic systems like [BibTeX][bibtex].
+    Some projects also have a separate `CONTRIBUTORS` file listing everyone who has contributed to it,
+    while others include this as a section in `CITATION`.
+
+If setting up the project or contributing to it are complex,
+there may also be a file called `BUILD` that explains how to install the required software,
+the formatting conventions for new data entries,
+and so on.
+These instructions can also be included as a section in `README`.
+Whichever convention is used,
+remember that the easier it is for people to get set up and contribute,
+the more likely they are to do so @Stei2014.
+
+TODO: `README` or `CONTRIBUTING` are also great places to document the conventions of structure (the rest of this chapter), naming and style.
+
+## How do I document a project?
+
+TODO: What does someone need to know beyond the collection of files in the project? 
+ 
+TODO: What should go in a `README`? Use Elizabeth's Step 4: "Administrative and personnel details" and "Project information" prompts as a guide. https://www.ideals.illinois.edu/bitstream/handle/2142/91611/W2%20-%20DataDocumentationHandout.docx
+
+## How should I structure the files and directories in my project? 
+
+TODO: "There are many strongly held opinions for how this should be done. These opinions are strongly held not becasue the authors think their way is **the** right way, but because its more important to be consistent, and it's easier to be consistent when you have publicly decreed that *this is* the way you will do something.  Does it matter which you use? Not really.  More important to use a structure that works for you, your audience and your project, be consistent, and communicate your conventions."
```

Comment:
I think we include something that mentions that although there is no one project structure to rule them all, there are guiding principles that exists in many if not all great project structures, e.g. make a folder for raw data and don't touch it.